docs/competing_risks.md

```diff
@@ -0,0 +1,278 @@
+	            else:
+	                mdata.loc[dex2,'outcome'] = "Change of government type" # could alternatively use dex1
+	
+	# The outcome variable must be a category type
+	mdata['outcome'] = mdata['outcome'].astype('category')
+
+We can do additional feature engineering such as including features that count the number of new leaders and changes of government types in the last 10 and 20 years.
+
+.. code-block:: python
+
+	# Feature engineering (optional)
+	# Include number of new leaders in last 10 and 20 years
+	# Include number of new governments types in last 10 and 20 years
+	mdata['new_leader_10'] = nan
+	mdata['new_leader_20'] = nan
+	mdata['new_govt_10'] = nan
+	mdata['new_govt_20'] = nan 
+	
+	for i in mdata['country'].unique():
+	    dex4 = mdata['country'] == i
+	    for j in mdata.loc[dex4,'year'].unique():
+	        if mdata.loc[dex4,'year'].min() <= j - 10:
+	           mdata.loc[dex4 & (mdata['year'] == j), 'new_leader_10'] = mdata.loc[dex4 & (mdata['year'] >= j - 10) & (mdata['year'] < j), 'country-leader-spell'].nunique()
+	           mdata.loc[dex4 & (mdata['year'] == j), 'new_govt_10'] = mdata.loc[dex4 & (mdata['year'] >= j - 10) & (mdata['year'] < j), 'government'].nunique()
+	           if mdata.loc[dex4,'year'].min() <= j - 20:
+	               mdata.loc[dex4 & (mdata['year'] == j), 'new_leader_20'] = mdata.loc[dex4 & (mdata['year'] >= j - 20) & (mdata['year'] < j), 'country-leader-spell'].nunique()
+	               mdata.loc[dex4 & (mdata['year'] == j), 'new_govt_20'] = mdata.loc[dex4 & (mdata['year'] >= j - 20) & (mdata['year'] < j), 'government'].nunique()
+	            
+	# We no longer need country-leader-spell 
+	mdata = mdata.drop(['country-leader-spell'], axis = 1)
+
+#### Training and forecasting
+
+##### Survival probabilities
+
+Now that the data is prepared, we can train and forecast with the survival modeler, `LGBSurvivalModeler`, to obtain the survival probabilities. The survival probabilities are defined to be 
+
+.. math::
+
+   Pr(T_i\geq t|X_{i}).
+
+Forecasts are produced for censored individuals with `.forecast()` up to the furthest observed time period after the initial period, :math:`t\in\{T_M+1,T_M+2,...,2\cdot T_M\}`. The features used are from the last observed row, :math:`X_{i{T_M}}`. Thus all forecasted probabilities are conditional on the same observed features and are coherent up to numerical accuracy. If a terminal event type column is in the data, it should not be passed to `LGBSurvivalModeler()`.
+
+.. code-block:: python
+
+	# Obtain the survival probability
+	survival_modeler = LGBSurvivalModeler(data=mdata.drop('outcome', axis = 1))
+	survival_modeler.build_model(parallelize=False)
+	survival_modeler_forecasts = survival_modeler.forecast()
+	
+	# Set custom column names
+	custom_forecast_headers = date_range(data["year-month"].max(),
+	                                      periods=len(survival_modeler_forecasts .columns),
```

Comment:
You should add 1 to the number of periods, then exclude the initial element of the resulting date_range. Then the range will start at August 2020 instead of July 2020. See [this example notebook](https://nbviewer.jupyter.org/github/IDA-HumanCapital/fife/blob/master/examples/country_leadership.ipynb#forecasts) (ctrl+F for "[8]").

Reply:
Done